### PR TITLE
fix: providing stable ordering for deprecation metadata

### LINKED ISSUE #34858: Deprecated CLI options and new options are not stable in their sorting
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dist/quarkus

### Describe the bug

Keycloak uses a set of the deprecated options and their alternatives, which leads to differently sorted output in the docs.

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

All options should be stable in their sorting. Possibly sorted alphabetically automatically.

### Actual behavior

The sorting switches between

```
STRICT and WILDCARD have been deprecated, use DEFAULT instead.
*Deprecated values: `+WILDCARD+`, `+STRICT+`*
```

and 

```
STRICT and WILDCARD have been deprecated, use DEFAULT instead.
*Deprecated values: `+STRICT+`, `+WILDCARD+`*
```

in the nightly build of Nov 9. 

### How to Reproduce?

See downstream docs build for RHBK26

### Anything else?

The `options.adoc` template or possibly one of the beans where the data is retrieved from needs fixing.

### PR DESCRIPTION
**Pull request by shawkins**
closes: #34858

### CODE REVIEW

File: quarkus/config-api/src/main/java/org/keycloak/config/DeprecatedMetadata.java [hunk 1]
```diff
@@ -17,21 +17,22 @@
...
 public class DeprecatedMetadata {
-    private final Set<String> newOptionsKeys;
+    private final Collection<String> newOptionsKeys;
     private final String note;
-    private final Set<String> deprecatedValues;
+    private final Collection<String> deprecatedValues;
 
     private DeprecatedMetadata(Set<String> newOptionsKeys, String note, Set<String> deprecatedValues) {
-        this.newOptionsKeys = newOptionsKeys == null ? Collections.emptySet() : Collections.unmodifiableSet(newOptionsKeys);
+        this.newOptionsKeys = newOptionsKeys == null ? Collections.emptySet() : newOptionsKeys.stream().sorted().toList();
```
[the remark this answers is not in the document]
Comment (shawkins) [reply quarkus/config-api/src/main/java/org/keycloak/config/DeprecatedMetadata.java:33]: To keep the original intent of immutablity, it doesn't save us anything to use a treeset.

[the remark this answers is not in the document]
Comment (mabartos) [reply quarkus/config-api/src/main/java/org/keycloak/config/DeprecatedMetadata.java:33]: ok

**Review by mabartos — APPROVED** [review]

**Review by mabartos — CHANGES_REQUESTED** [review]
@shawkins It would be good to add some simple test.

**Review by mabartos — APPROVED** [review]

**Review by mabartos — CHANGES_REQUESTED** [review]
Oops, the test xD 

I see that you added changes regarding the PicocliTest package, so I can close this PR: https://github.com/keycloak/keycloak/pull/35146

**Comment by shawkins** [comment]
> Oops, the test xD
> 
> I see that you added changes regarding the PicocliTest package, so I can close this PR: #35146

Right, I had just noticed that as well. A test has been added to this pr as well - it was missing from the last commit.

**Review by vmuzikar — APPROVED** [review]
LGTM, thanks for the fix!

**Review by mabartos — APPROVED** [review]
LGTM as well

**Pull request merged**